src/EventStore.Core/Services/Transport/Http/Controllers/PersistentSubscriptionController.cs

```diff
@@ -63,6 +65,116 @@ public class PersistentSubscriptionController : CommunicationController {
+			var stream = match.BoundVariables["stream"];
+			var groupname = match.BoundVariables["group"];
+			var evNum = match.BoundVariables["event"];
+			var cnt = match.BoundVariables["count"];
+
+			stream = "$persistentsubscription-" + stream + "::" + groupname+"-parked";
+			
+			
+			long eventNumber = -1;
+			int count = AtomSpecs.FeedPageSize;
+			var embed = GetEmbedLevel(http, match);
+
+			if (stream.IsEmptyString()) {
+				SendBadRequest(http, string.Format("Invalid stream name '{0}'", stream));
+				return;
+			}
+
+			if (evNum != null && evNum != "head" && (!long.TryParse(evNum, out eventNumber) || eventNumber < 0)) {
+				SendBadRequest(http, string.Format("'{0}' is not valid event number", evNum));
+				return;
+			}
+
+			if (cnt.IsNotEmptyString() && (!int.TryParse(cnt, out count) || count <= 0)) {
+				SendBadRequest(http, string.Format("'{0}' is not valid count. Should be positive integer", cnt));
+				return;
+			}
+
+			bool resolveLinkTos;
+			if (!GetResolveLinkTos(http, out resolveLinkTos, true)) {
+				SendBadRequest(http, string.Format("{0} header in wrong format.", SystemHeaders.ResolveLinkTos));
+				return;
+			}
+
+			if (!GetRequireLeader(http, out var requireLeader)) {
+				SendBadRequest(http, string.Format("{0} header in wrong format.", SystemHeaders.RequireLeader));
+				return;
+			}
+
+			bool headOfStream = eventNumber == -1;
+			GetStreamEventsBackward(http, stream, eventNumber, count, resolveLinkTos, requireLeader, headOfStream,
+				embed);
+		}
+		private void GetStreamEventsBackward(HttpEntityManager manager, string stream, long eventNumber, int count,
+			bool resolveLinkTos, bool requireLeader, bool headOfStream, EmbedLevel embed) {
+			var envelope = new SendToHttpEnvelope(_networkSendQueue,
+				manager,
+				(ent, msg) =>
+					Format.GetStreamEventsBackward(ent, msg, embed, headOfStream),
+				(args, msg) => Configure.GetStreamEventsBackward(args, msg, headOfStream));
+			var corrId = Guid.NewGuid();
+			Publish(new ClientMessage.ReadStreamEventsBackward(corrId, corrId, envelope, stream, eventNumber, count,
+				resolveLinkTos, requireLeader, GetETagStreamVersion(manager), manager.User));
+		}
+		private bool GetRequireLeader(HttpEntityManager manager, out bool requireLeader) {
+			requireLeader = false;
+			
+			var onlyLeader = manager.HttpEntity.Request.GetHeaderValues(SystemHeaders.RequireLeader);
+			var onlyMaster = manager.HttpEntity.Request.GetHeaderValues(SystemHeaders.RequireMaster);
+			
+			if (StringValues.IsNullOrEmpty(onlyLeader) && StringValues.IsNullOrEmpty(onlyMaster))
+				return true;
+		
+			if (string.Equals(onlyLeader, "True", StringComparison.OrdinalIgnoreCase) ||
+			    string.Equals(onlyMaster, "True", StringComparison.OrdinalIgnoreCase)) {
+				requireLeader = true;
+				return true;
+			}
+
+			return string.Equals(onlyLeader, "False", StringComparison.OrdinalIgnoreCase) ||
+			       string.Equals(onlyLeader, "False", StringComparison.OrdinalIgnoreCase);
```

Comment:
The following should likely be...

```suggestion
			       string.Equals(onlyMaster, "False", StringComparison.OrdinalIgnoreCase);
```

We can check the AtomController as well and either fix it in this PR or raise an issue to ensure that we don't have more of this logic lying around.

Reply:
Thanks! Looks like they were only in Admin and Atom controllers. Made the fixes.